lib/rubocop/cop/chef/property_with_name_attribute.rb

```diff
@@ -0,0 +1,59 @@
+#
+# Copyright:: Copyright 2019, Chef Software Inc.
+# Author:: Tim Smith (<tsmith@chef.io>)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+module RuboCop
+  module Cop
+    module Chef
+      # When using properties in a custom resource you should use name_property not
+      # the legacy name_attribute from the days of attributes
+      #
+      # @example
+      #
+      #   # bad
+      #   property :bob, String, name_attribute: true
+      #
+      #   # good
+      #   property :bob, String, name_property: true
+      #
+      class PropertyWithNameAttribute < Cop
+        MSG = 'Resource property sets name_attribute not name_property'.freeze
+
+        def on_send(node)
+          add_offense(node, location: :expression, message: MSG, severity: :refactor) if attribute_method_mix?(node)
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            corrector.replace(node.loc.expression, node.source.gsub('name_attribute: true', 'name_property: true'))
```

Comment:
/name_attribute:\s*true/ at least here

Reply:
there's probably a slicker way to match with the rubocop AST "regexp" on the hash arguments and "capture" only the `name_attribute: true` bit as an AST object and then replace that in a way that wouldn't have ruby-parsing vs. string-parsing edge conditions.  but this probably works for now.

Reply:
I just switched it to gsub on name_attribute and name_property since that's all we actually care about

Reply:
#126 tracks doing better autocorrects though